pandas/tseries/tests/test_base.py

```diff
@@ -504,6 +504,29 @@ def test_infer_freq(self):
             self.assertEqual(result.freq, freq)
 
 
+def test_datetimelike_tostring_astype():
+    result = pd.date_range('2012-01-01', periods=4, name='test_name').astype(str)
+    expected = pd.Index(['2012-01-01', '2012-01-02', '2012-01-03','2012-01-04'],
+                        name='test_name', dtype=object)
+    tm.assert_index_equal(result, expected)
+
+
+def test_datetimelike_tostring_astype_freqH():
+    result = pd.date_range('1/1/2011', periods=3, freq='H', name='test_name').astype(str)
+    expected = pd.Index(['2011-01-01 00:00:00', '2011-01-01 01:00:00', '2011-01-01 02:00:00'], name='test_name',
+                        dtype=object)
+    tm.assert_index_equal(result, expected)
+
+
+def test_datetimelike_tostring_astype_freqH_timezone():
+    # _skip_if_no_pytz()
+    import pytz
+    tz_pytz = pytz.timezone('Europe/London')
+    result = pd.date_range('3/6/2012 00:00', periods=2, freq='H', tz=tz_pytz, name='test_name').astype(str)
+    expected = pd.Index(['2012-03-06 00:00:00+00:00', '2012-03-06 01:00:00+00:00'], dtype=object, name='test_name')
+    tm.assert_index_equal(result, expected)
```

Comment:
these can all be in a single test, e.g. ``test_astype_str`` (since its inside the ``TestDatetimeIndexOps`` namespace